packages/caip/src/adapters/thortrading/index.ts

```diff
@@ -0,0 +1,70 @@
+import { fromAssetId, toAssetId } from '../../assetId/assetId'
+import { btcAssetId, ethAssetId, ethChainId } from '../../constants'
+import { AssetId } from './../../assetId/assetId'
+import { btcChainId } from './../../constants'
+
+// https://regex101.com/r/d3Lgl9/1
+const thorIdRegex =
+  /(?<THORChainChain>[A-Z]+)\.(?<symbol>[A-Z]+)(-(?<contractAddress>[A-Z0-9]{42}))?/
+
+/*
+ * Note: this only works with thorchain assets we support
+ * see https://dev.thorchain.org/thorchain-dev/memos#asset-notation for reference
+ */
+export const poolAssetIdToAssetId = (id: string): AssetId | undefined => {
+  const matches = thorIdRegex.exec(id)?.groups
+  const THORChainChain = matches?.THORChainChain
+  const symbol = matches?.symbol
+
+  switch (THORChainChain) {
+    case 'ETH': {
+      if (id === 'ETH.ETH') return ethAssetId
```

Comment:
is there a specific edge case this catches? I would think this would all be derived from the matches directly and not add the `id` string back into the mix